aptos-labs/aptos-core
# brian/proof manager concurrent tests

- [Quorum Store] BatchGenerator produces multiple bucketed batches and multi-batch messages are used to send, vote, and broadcast.
- [TEST ONLY] turn on quorum store for testing
- [TEST ONLY] three_region_simulation_graceful_overload
- Add buckets to the metrics
- ensure that multi-batch messages are not empty
- BatchGenerator cuts txns into max batch sizes
- Bucket is downgraded if inflight txns have smaller gas
- Fix bug on last bucket, with unit test
- Address minor comments
- Allow resubmitted high gas transactions to get into Quorum Store batches. No other provisions are made. This means if the user submits a high gas transaction with low gas transactions in flight, the high gas transaction will likely retry after failing in consensus execution.
- Revert (semantically) 'Bucket is downgraded if inflight txns have smaller gas'
- Address comments in batch_generator
- Add sender and receiver limits on BatchMsg.
- Verify the length of the multi-batch vectors, before other verification
- [Quorum Store] fairness in proposal via per-author queue in proof manager
- Add an expiration index, can probably add some methods to make all the keys a bit less ugly
- Make the nonce use the system time on DB startup
- Use the existing TimeExpirations instead of a new BTreeMap
- Address comment
- Reduce backlog_txn_limit_count because consensus behaves better
- three_region_simulation

### Description

### Test Plan
